src/ast.rs

```diff
@@ -79,6 +79,8 @@ impl<'i> CoreParser<'i> {
       _ if self.try_consume("&")   => hvm::OP_AND,
       _ if self.try_consume("|")   => hvm::OP_OR,
       _ if self.try_consume("^")   => hvm::OP_XOR,
+      _ if self.try_consume("sin") => hvm::OP_SHL,
```

Comment:
These symbols are trying to uniquely map to operators, so we shouldn't add sin and tan here. Notice how there's no pow, log and atan2.

Reply:
fair enough but in the context of `[` nothing else could appear there, thought it would be a bit more readable.